ten-simple-rules-dockerfiles.Rmd

```diff
@@ -83,8 +88,7 @@ The differences between a helpful, stable `Dockerfile` and one that is misleadin
 A commitment to this article's rules can ensure that workflows are reproducible and reusable, that computing environments are understandable by others, and researchers can collaborate effectively.
 Their application should not be triggered by the publication of a finished project, but be weaved into day-to-day habits (cf. thoughts on openness as an afterthought by @chen_open_2019 and on computational reproducibility by @donoho_invitation_2010).
 
-To start with, we assume you have a scripted scientific workflow, i.e. you can, at least at a certain point in time, execute the full process with a single command, for example `make my_workflow`, `Rscript analysis.R`, or `python3 paper-plots.py`.
-This execution should be able to be triggered by command-line instruction, which is possible for all generic programming languages and widely used workflow tools, but may also be opening and starting a process with a graphical user interface.
+To start with, we assume you have a scripted scientific workflow, i.e. you can, at least at a certain point in time, execute the full process with a single command, for example `make my_workflow`, `Rscript analysis.R`, or `python3 paper-plots.py`. (A key constraint is Docker containers can only run open source software, so environments like Mathematica and Matlab cannot currently be easily used within containers.) This execution should be able to be triggered by command-line instruction, which is possible for all generic programming languages and widely used workflow tools, but may also be opening and starting a process with a graphical user interface.
```

Comment:
hmm, in my experience most (what I would consider robust scientific workflows) don't get executed with one python or R script. It's more likely to see:

```
snakemake --use-conda <other params>
nextflow workflow.nf --in 'dataset/*.fa'
java -jar cromwell-XY.jar run myWorkflow.wdl
```
At least for workflows with sophisticted DAGS, we've gone well beyond `<executable> <run my script>` - the above abstracts away / misleads that there isn't complexity in suggesting that workflows are started like that. See https://docs.google.com/spreadsheets/d/1plkAsT_S3CzSeb7ivxyjRnHyrK3JclUCXeUMf_azraY/edit#gid=0

Reply:
We also contradict ourselves by saying it opens a GUI but then directly after that GUIs are not reproducible.

Reply:
Thanks for all your comments today, v. helpful.
I agree that many workflows may use a variety of tools to assemble the results.  Having an example Dockerfile showing that would be good.  (Most of mine tend to be at the simpler end, with just some R scripts that need running.)